explorer/testdata/function/fail_recurse_before_typecheck.carbon

```diff
@@ -0,0 +1,19 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{not} %{explorer} %s 2>&1 | %{FileCheck-strict} %s
+// RUN: %{not} %{explorer-trace} %s 2>&1 | %{FileCheck-allow-unmatched} %s
+// AUTOUPDATE: %{explorer} %s
+
+package ExplorerTest api;
+
+fn T() -> Type {
```

Comment:
Can you also test `fn T() -> T() { return i32; }`?

I expect that'll correctly fail, but it's a different recursion path.

Reply:
Added function/fail_recurse_in_return_type.carbon